File: fib2.lua
-- lua5.2 runs fib(40) in ~58 seconds
-- luajit runs fib(40) in ~10.2 seconds

function fib(args)
    local n = args.n
    if n < 2 then return n end
    return fib{n=n - 1} + fib{n=n - 2}
end

print(fib{n=40})

File: isrc/fib2.py
# python3 fib(40) runs in ~ 45.8 s.
# pypy3 fib(40) runs in ~ 2.86 s.

def fib(**args):
    n = args['n']
    if n < 2:
        return n
    return fib(n=n - 1) + fib(n=n - 2)

print(fib(n=40))
